src/server/pps/server/api_server.go

```diff
@@ -1979,13 +1952,22 @@ func (a *apiServer) hardStopPipeline(pachClient *client.APIClient, pipelineInfo
 		}
 		if ci.Finished == nil {
 			// Finish the commit and don't pass a tree
-			pachClient.PfsAPIClient.FinishCommit(pachClient.Ctx(), &pfs.FinishCommitRequest{
+			txnCtx.Pfs().FinishCommitInTransaction(txnCtx, &pfs.FinishCommitRequest{
```

Comment:
This isn't really a 'safe' operation, in my opinion, because the set of commits might change because the `ListCommitStream` is done outside of the transaction.  This may need to use the commits collection directly, although it would be kind of ugly to do that from PPS.  Also (just something to think about, not sure if it's actually a good idea), but would this function make sense to do deferred to the end of the transaction the same way we do `PropagateCommit` in PFS - we could append a similar task to be performed at the end, and let all the code for that stay in PFS.